docs/reference/ingest/enrich.asciidoc

```diff
@@ -53,20 +123,359 @@ that doesn't change frequently.
 
-To begin,
-create one or more source indices.
+image::images/ingest/enrich/enrich-process.svg[align="center"]
 
-A _source index_ contains data you want to append to incoming documents.
-You can index and manage documents in a source index
-like a regular index.
+Before implementing the pipeline in production, we recommend indexing a few test
+documents first and verifying enrich data was added correctly using the
+<<docs-get,get API>>.
+
+[[update-enrich-data]]
+==== Update an enrich index
+
+include::{docdir}/ingest/apis/enrich/execute-enrich-policy.asciidoc[tag=update-enrich-index]
+
+If wanted, you can <<docs-reindex,reindex>>
+or <<docs-update-by-query,update>> any already ingested documents
+using your ingest pipeline.
+
+[[update-enrich-policies]]
+==== Update an enrich policy
+
+include::apis/enrich/put-enrich-policy.asciidoc[tag=update-enrich-policy]
+
+[role="xpack"]
+[testenv="basic"]
+[[enrich-policy-definition]]
+=== Enrich policy definition
 
-The following <<docs-index_,index API>> request creates the `users` source index
-containing user data.
-This request also indexes a new document to the `users` source index.
+<<enrich-policy,Enrich policies>> are defined as JSON objects like the
+following:
+
+[source,js]
+----
+{
+    "<enrich_policy_type>": {
+        "indices": ["..."],
+        "match_field": "...",
+        "enrich_fields": ["..."],
+        "query": "..."
+    }
+  }
+}
+----
+// NOTCONSOLE
+
+[[enrich-policy-parms]]
+==== Parameters
+
+`<enrich_policy_type>`::
++
+--
+(Required, enrich policy object)
+The enrich policy type determines how enrich data is matched to incoming
+documents.
+
+Supported enrich policy types include:
+
+<<geo-match-enrich-policy-type,`geo_match`>>:::
+Matches enrich data to incoming documents based on a geographic location using
+a <<query-dsl-geo-shape-query,`geo_shape` query>>. For an example, see
+<<geo-match-enrich-policy-type>>.
+
+<<match-enrich-policy-type,`match`>>:::
+Matches enrich data to incoming documents based on a precise value, such as an
+email address or ID, using a <<query-dsl-term-query,`term` query>>. For an
+example, see <<match-enrich-policy-type>>.
+--
+
+`indices`::
++
+--
+(Required, String or array of strings)
+Source indices used to create the enrich index.
+
+If multiple indices are provided, they must share a common `match_field`, which
+the enrich processor can use to match incoming documents.
+--
+
+`match_field`::
+(Required, string)
+Field in the source indices used to match incoming documents.
+
+`enrich_fields`::
+(Required, Array of strings)
+Fields to add to matching incoming documents. These fields must be present in
+the source indices.
+
+`query`::
+(Optional, string)
+Query type used to filter documents in the enrich index for matching. Valid
+value is <<query-dsl-match-all-query,`match_all`>> (default).
+
+[role="xpack"]
+[testenv="basic"]
+[[geo-match-enrich-policy-type]]
+=== Geo match enrich policy type
```

Comment:
would it be useful to highlight that this is a good place to go for seeing an example of its usage? I'm thinking that when I look at the table-of-contents, it would be nice to give a more human-friendly description of what I am going to find when I click on `Geo match enrich policy type` or `Match enrich policy type`. I would be delighted to know that if I am looking at how to use or read an example, it would be stated in this table. 

Might be overkill, so feel free to ignore.

Reply:
I don't feel this is overkill at all. I actually went through several iterations with naming these (important) pages. I'm glad you highlighted this.

With d09f0dc43ff383e9ff7d3c32d3b29bec0ed13ffa, I changed the titles to include `Example` and tried to use human-centric language. I also added some cross-links if someone lands here without first seeing the overview content.

I really value any other feedback you have here so let me know if these new titles don't work!

Reply:
saweeet!